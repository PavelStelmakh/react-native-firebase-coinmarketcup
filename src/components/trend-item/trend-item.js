import React from 'react';
import Icon from 'react-native-vector-icons/Entypo';
import AntIcon from 'react-native-vector-icons/AntDesign';
import { View, Image, Text, TouchableHighlight } from 'react-native';
import { TrendChart } from '../trend-chart';
import { formatCurrency, formatPercent } from '../../utils/number-formatting';
import { styles } from './trend-item.style';

const trendsUpIconProps = {
  name: 'triangle-up',
  color: '#16c784',
};

const trendsDownIconProps = {
  name: 'triangle-down',
  color: '#ea3943',
};

export const TrendItem = ({
  logo,
  name,
  number,
  symbol,
  percentChange,
  price,
  marketCapital,
  marked,
  onMark,
  onOpenDetails,
}) => (
  <TouchableHighlight onPress={onOpenDetails}>
    <View style={styles.container}>
      <Image source={{ uri: logo }} style={styles.image} />
      <View style={styles.currencyInfoContainer}>
        <View style={styles.currencyNameInfoContainer}>
          <Text style={[styles.text, styles.title]}>{name}</Text>
          <View style={styles.flexContainer}>
            <View style={styles.flexContainer}>
              <View style={styles.itemNumberContainer}>
                <Text style={[styles.itemNumber, styles.text]}>{number}</Text>
              </View>
              <Text style={styles.subTitle}>{symbol}</Text>
            </View>
            <View style={[styles.flexContainer, styles.indexContainer]}>
              <Icon
                {...(percentChange >= 0
                  ? trendsUpIconProps
                  : trendsDownIconProps)}
                size={15}
              />
              <Text style={styles.subTitle}>
                {formatPercent(Math.abs(percentChange))}
              </Text>
            </View>
          </View>
        </View>
        <TrendChart
          containerStyle={styles.chartContainer}
          trendPositive={percentChange >= 0}
          assetKey={name.replace(/\s/gi, '-').toLowerCase()}
        />
        <View style={styles.currencyAmount}>
          <Text style={[styles.text, styles.title]}>
            {formatCurrency(price)}
          </Text>
          <Text style={styles.subTitle}>
            Mar. cup. {formatCurrency(marketCapital)}
          </Text>
        </View>
      </View>
      <View style={styles.trendMark}>
        <AntIcon
          name={marked ? 'star' : 'staro'}
          color={marked ? '#f6b87e' : 'rgb(133, 140, 162)'}
          size={10}
          onPress={() => onMark(!marked)}
        />
      </View>
    </View>
  </TouchableHighlight>
);
