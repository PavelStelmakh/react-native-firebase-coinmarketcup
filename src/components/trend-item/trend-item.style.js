import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  container: {
    paddingVertical: 10,
    paddingLeft: 10,
    flexDirection: 'row',
    borderBottomWidth: 0.2,
    borderColor: 'rgba(133, 140, 162, 0.5)',
  },
  flexContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  title: {
    fontWeight: 'bold',
    fontSize: 13,
    marginBottom: 2,
  },
  subTitle: {
    fontSize: 10,
    color: 'rgb(133, 140, 162)',
  },
  image: {
    width: 27,
    height: 27,
    marginRight: 7,
    flexGrow: 0,
    flexShrink: 0,
  },
  currencyInfoContainer: {
    flexGrow: 1,
    flexDirection: 'row',
  },
  currencyNameInfoContainer: {
    flexGrow: 3,
    flexBasis: 0,
  },
  indexContainer: {
    marginLeft: 5,
  },
  chartContainer: {
    flexGrow: 1,
    flexShrink: 0,
    flexBasis: 50,
    alignItems: 'center',
  },
  currencyAmount: {
    alignItems: 'flex-end',
    flexGrow: 3,
    flexBasis: 0,
  },
  text: {
    color: '#fff',
  },
  itemNumberContainer: {
    paddingHorizontal: 5,
    height: 13,
    borderRadius: 4,
    backgroundColor: 'rgb(133, 140, 162)',
    marginRight: 3,
  },
  itemNumber: {
    fontSize: 10,
    lineHeight: 13,
  },
  trendMark: {
    alignItems: 'center',
    justifyContent: 'center',
    width: 30,
  },
});
