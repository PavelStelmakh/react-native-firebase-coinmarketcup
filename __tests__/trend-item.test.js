import 'react-native';
import React from 'react';
import { render, screen } from '@testing-library/react-native';
import { TrendItem } from '../src/components/trend-item';

describe('<TrendItem />', () => {
  it('should rendercomponent', () => {
    const props = {
      logo: 'https://logo.com/btc',
      name: 'Bitcoin',
      number: 1,
      symbol: 'BTC',
      percentChange: 21,
      price: 1234,
      marketCapital: 123456789123,
      marked: false,
      onMark: jest.fn(),
      onOpenDetails: jest.fn(),
    };

    render(<TrendItem {...props} />);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
