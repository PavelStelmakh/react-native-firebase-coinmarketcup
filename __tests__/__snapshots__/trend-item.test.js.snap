// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrendItem /> should rendercomponent 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Object {
        "borderBottomWidth": 0.2,
        "borderColor": "rgba(133, 140, 162, 0.5)",
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingVertical": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://logo.com/btc",
        }
      }
      style={
        Object {
          "flexGrow": 0,
          "flexShrink": 0,
          "height": 27,
          "marginRight": 7,
          "width": 27,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexGrow": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexBasis": 0,
            "flexGrow": 3,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
              },
              Object {
                "fontSize": 13,
                "fontWeight": "bold",
                "marginBottom": 2,
              },
            ]
          }
        >
          Bitcoin
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgb(133, 140, 162)",
                  "borderRadius": 4,
                  "height": 13,
                  "marginRight": 3,
                  "paddingHorizontal": 5,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 10,
                      "lineHeight": 13,
                    },
                    Object {
                      "color": "#fff",
                    },
                  ]
                }
              >
                1
              </Text>
            </View>
            <Text
              style={
                Object {
                  "color": "rgb(133, 140, 162)",
                  "fontSize": 10,
                }
              }
            >
              BTC
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "marginLeft": 5,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#16c784",
                    "fontSize": 15,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Entypo",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "rgb(133, 140, 162)",
                  "fontSize": 10,
                }
              }
            >
              21,000 %
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexBasis": 50,
            "flexGrow": 1,
            "flexShrink": 0,
          }
        }
      >
        <ActivityIndicator />
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexBasis": 0,
            "flexGrow": 3,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
              },
              Object {
                "fontSize": 13,
                "fontWeight": "bold",
                "marginBottom": 2,
              },
            ]
          }
        >
          $1 234 
        </Text>
        <Text
          style={
            Object {
              "color": "rgb(133, 140, 162)",
              "fontSize": 10,
            }
          }
        >
          Mar. cup. 
          $123,457 Bn
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 30,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        selectable={false}
        style={
          Array [
            Object {
              "color": "rgb(133, 140, 162)",
              "fontSize": 10,
            },
            undefined,
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
